### PR TITLE
Feature/220

### LINKED ISSUE #220: Add organizations page to volunteer toronto routes
https://xd.adobe.com/view/4a36317f-0fd9-444a-b1cd-e135d91ce2e0-1f65/screen/696def73-8409-47db-ad71-4b82cc589af7?fullscreen&hints=off

Add organizations page to /pages/volunteer-toronto/organizations

Should follow designs in /pages/volunteer-toronto/members

Reviewer: @WendyChenj

<|<- META INFO (DO NOT MODIFY CONTENTS MANUALLY) ->|>
*BLOCKED BY*

- [ ] #236

*BLOCKING*

- [ ] #136

<|<- META INFO END ->|>

### PR DESCRIPTION
**Pull request by sidprak**
Resolves #220

### CODE REVIEW

File: web-client/src/pages/api/vtmembers/mockMember.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import {MembersListItem} from '../../../shared/redux/volunnteer-toronto/members'
+import {MembersListItem} from '../../../shared/redux/volunteer-toronto/members'
```
Comment (l2silver) [inline web-client/src/pages/api/vtmembers/mockMember.ts:1]: Nice catch!

File: web-client/src/shared/components/volunteer-toronto/members/MemberTable/ButtonGroup.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 import React, {useState} from 'react'
-import { MembersListItem, deleteMember } from '../../../../redux/volunnteer-toronto/members'
+import { MembersListItem, deleteMember } from '../../../../redux/volunteer-toronto/members'
 import ConfirmDeleteDialog from './ConfirmDeleteDialog'
```
Comment (l2silver) [inline web-client/src/shared/components/volunteer-toronto/members/MemberTable/ButtonGroup.tsx:4]: This is embarrassing

File: web-client/src/shared/components/volunteer-toronto/organizations/OrganizationTable/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+} from '@material-ui/core'
+import { useSelector, useDispatch } from 'react-redux'
+
+const OrganizationsTable:React.FunctionComponent = ()=>{
+    const dispatch = useDispatch()
+    const loading = useSelector(loadingSelector)
+    const orgs = useSelector(filteredOrganizationListSelector)
+
+    useEffect(()=>{
+        dispatch(fetchOrganizations())
```
Comment (l2silver) [inline web-client/src/shared/components/volunteer-toronto/organizations/OrganizationTable/Table.tsx:28]: Looks like you've mastered hooks. Nice work

File: web-client/src/shared/redux/volunteer-toronto/organizations/index.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    setLoading
+} = volunteerTorontoOrganizationsSlice.actions
+
+export const fetchOrganizations = ():OrganizationThunk =>{
+    return async (dispatch)=>{
+        dispatch(setLoading(true))
+        try{
+            const host = process.env.NEXT_PUBLIC_API_SERVER_URL
+            const res = await fetch(`${host}/api/organizations`)
+            const organizations = await res.json()
```
Comment (l2silver) [inline web-client/src/shared/redux/volunteer-toronto/organizations/index.ts:54]: This looks good! I think in the future, we'll probably do a check for the status code of the response, and add an error value to the reducer that we can set in the catch statement, but this is more than enough for now.

File: web-client/src/shared/redux/volunteer-toronto/organizations/index.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        try{
+            const host = process.env.NEXT_PUBLIC_API_SERVER_URL
+            const res = await fetch(`${host}/api/organizations`, {
+                method: 'POST',
+                headers: {'Content-Type': 'application/json'},
+                body:JSON.stringify({id:organizationId})
+            })
+
+            const confirmation = await res.json()
+            if(confirmation.data.id){
```
Comment (l2silver) [inline web-client/src/shared/redux/volunteer-toronto/organizations/index.ts:79]: Nice

File: web-client/src/shared/redux/volunteer-toronto/organizations/index.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    return (dispatch, getState)=>{
+        const state = getState().volunteerTorontoOrganizations
+        console.log("filter: " + filterString);
+        if(filterString === ''){
+            dispatch(setFilteredOrganizations(state.organizations))
+        }
+        else{
+            dispatch(setFilteredOrganizations(
+                state.organizations.filter(organization=>(
+                    new RegExp(`${filterString}`,'i').test(organization.name)
```
Comment (l2silver) [inline web-client/src/shared/redux/volunteer-toronto/organizations/index.ts:104]: Very clean

**Review by l2silver — APPROVED** [review]
I was a little scared with 18 file changes, but this looks good to go!

File: web-client/src/shared/components/volunteer-toronto/organizations/OrganizationTable/Table.tsx (hunk 3 above)
Comment (sidprak) [reply web-client/src/shared/components/volunteer-toronto/organizations/OrganizationTable/Table.tsx:28]: 😄 I took a lot of inspiration from the members page.

File: web-client/src/shared/redux/volunteer-toronto/organizations/index.ts (hunk 4 above)
Comment (sidprak) [reply web-client/src/shared/redux/volunteer-toronto/organizations/index.ts:54]: Sounds good!

**Pull request merged**